core/components/handler/handler.go

```diff
@@ -461,31 +461,28 @@ func (h component) consumeJoin(appEUI []byte, devEUI []byte, appKey [16]byte, da
 	}
 	packet := bundles[best.ID].Packet.(*core.JoinHandlerReq)
 
-	// Generate a DevAddr + NwkSKey + AppSKey
-	ctx.Debug("Generate DevAddr, NwkSKey and AppSKey")
+	// Generate a DevAddr - Note: this should be done by the Broker (issue #90). Random generation should be moved to the random package
 	rdn := rand.New(rand.NewSource(int64(packet.Metadata.Rssi)))
-	appNonce, devAddr := make([]byte, 4), [4]byte{}
-	binary.BigEndian.PutUint32(appNonce, rdn.Uint32())
+	var devAddr [4]byte
 	binary.BigEndian.PutUint32(devAddr[:], rdn.Uint32())
 	devAddr[0] = (h.Configuration.NetID[2] << 1) | (devAddr[0] & 1) // DevAddr 7 msb are NetID 7 lsb
 
-	buf := make([]byte, 16)
-	copy(buf[1:4], appNonce[:3])
-	copy(buf[4:7], h.Configuration.NetID[:])
-	copy(buf[7:9], packet.DevNonce)
+	// Generate appNonce - Note: this should be moved to the random package
```

Comment:
Move indeed, but at least fill the 7 MSB with the NwkID (7 LSB of NetID, this should come from configuration)